src/axolotl/utils/schemas/config.py

```diff
@@ -412,6 +416,76 @@ def check_gptq_w_revision(cls, data):
             )
         return data
 
+    @model_validator(mode="before")
+    @classmethod
+    def normalize_attn(cls, data):  # pylint: disable=too-many-return-statements
+        # cases where both are set and already match
+        if data.get("attention") == AttentionBackend.eager and data.get(
+            "eager_attention"
+        ):
+            return data
+        if data.get("attention") == AttentionBackend.flash and data.get(
+            "flash_attention"
+        ):
+            return data
+        if data.get("attention") == AttentionBackend.s2 and data.get("s2_attention"):
+            return data
+        if data.get("attention") == AttentionBackend.sdpa and data.get("sdp_attention"):
+            return data
+        if data.get("attention") == AttentionBackend.xformers and data.get(
+            "xformers_attention"
+        ):
+            return data
```

Comment:
Is this check needed?

I think this function could be re-arranged.

First, check for mismatch (third block -> first block). Then set attention if not done (second block). This first block can be removed?